src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorUpdateProjectParams.cs

```diff
@@ -1,24 +1,13 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System.Collections.Generic;
-using Microsoft.AspNetCore.Razor.Language;
 using Microsoft.AspNetCore.Razor.LanguageServer.Common.Serialization;
-using Microsoft.CodeAnalysis.Razor.ProjectSystem;
 using OmniSharp.Extensions.Embedded.MediatR;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem
 {
     internal class RazorUpdateProjectParams : IRequest
     {
-        public string FilePath { get; set; }
-
-        public RazorConfiguration Configuration { get; set; }
-
-        public string RootNamespace { get; set; }
-
-        public ProjectWorkspaceState ProjectWorkspaceState { get; set; }
-
-        public DocumentSnapshotHandle[] Documents { get; set; }
+        public FullProjectSnapshotHandle ProjectSnapshotHandle { get; set; }
```

Comment:
Before this we were relying on basic Newtonsoft deserialization to update this parent object and if we failed to properly deserialize it'd actually crash the language server. Therefore, we now let things flow through the "correct" deserialization path which is `FullProjectSnapshotHandle` and on a failure to deserialize this property is `null`.